src/Compilers/Core/CodeAnalysisTest/CryptoBlobParserTests.cs

```diff
@@ -3,21 +3,93 @@
 using Roslyn.Test.Utilities;
 using System;
 using System.Collections.Immutable;
+using System.Linq;
 using System.Diagnostics;
 using System.Globalization;
+using System.Security.Cryptography;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
     public class CryptoBlobParserTests : TestBase
     {
         [Fact]
+        public void GetPrivateKeyFromKeyPair()
+        {
+            var key = ImmutableArray.Create(TestResources.General.snKey);
+
+            RSAParameters? privKeyOpt;
+            Assert.True(CryptoBlobParser.TryParseKey(key, out _, out privKeyOpt));
+            Debug.Assert(privKeyOpt.HasValue);
+            var privKey = privKeyOpt.Value;
+
+            Assert.Equal(privKey.Exponent, new byte[] { 0x01, 0x00, 0x01, 0x00 });
+
+            var expectedModulus = key.Skip(20).Take(128).ToArray();
```

Comment:
Did you mean another method?  I don't see how sequential integers would be helpful here.